### ISSUE TITLE
Token only includes numbers

### ISSUE THREAD

**Issue by GrandpaVape** [gregod/addon-usbhid-to-mqtt#3]
I have barcodes with alphabetical characters and dashes. It seems this add-on automatically "chomps" off anything that isn't a number. Is there anyway to add an option do disable this and post the entire scanned code to MQTT?

**Comment by gregod** [comment]
My reader did only support numerical codes, hence the filter for those. Should be easy to extend, only varying keyboard layouts might complicate things. I will have a look this weekend.

**Comment by gregod** [comment]
I have just released a new version containing support for non-numeric characters. Could you please verify that it works as expected, especially in regard to the dashes.

**Comment by GrandpaVape** [comment]
Wow! Thank you for the quick response and attempt to fix!

I installed the updated version, but now it seems I'm just getting blank tokens now. My barcode scanner has 2 keyboard modes, "standard" and "virtual". With the previous version, I was getting only number when in standard, and blank tokens with virtual. Now I get no tokens with standard, and virtual gives me varying quantities of "LALT".

I am ready and willing to keep troubleshooting!

**Comment by GrandpaVape** [comment]
So I don't know if it was the update you made, or the fact that I might have been ignoring the max token length errors, but I increased the length, and I'm getting some results! Here are some codes I scanned on my PC paired with the log output in HASS.

```
BEP180-7
INFO:root:Token: LSHFTbLSHFTeLSHFTp1807

BE790C-3
INFO:root:Token: LSHFTbLSHFTe790LSHFTc3

BEP120-3
INFO:root:Token: LSHFTbLSHFTeLSHFTp1203

BEM540-7
INFO:root:Token: LSHFTbLSHFTeLSHFTm5407
```
Seems like it's grabbing all the keypresses (sans the '-' key). I noticed the [Stack Overflow question](https://stackoverflow.com/questions/19732978/how-can-i-get-a-string-from-hid-device-in-python-with-evdev/19757397#19757397) in [run.py](https://github.com/gregod/addon-usbhid-to-mqtt/blob/master/run.py#L83) and it seems there's a method of processing the shifts into capitals. I know absolutely nothing about Python, so I appreciate your patience in assisting me!

**Comment by gregod** [comment]
Ah yes! I only included support for lower caps. I will fix this and i think this will resolve your issue.

Just to make sure that there is nothing else going on: 
I have added a debug option in 1.0.1.   Could you enable this, set max_token length to 999, and send me the output for one of the codes above? I only need the lines starting with "DEBUG:root:USB Event"

**Comment by GrandpaVape** [comment]
Bingo! Every scan now reports a key event like this:

`WARNING:root:Unknown scancode (key event at 1619503871.225593, 74 (KEY_KPMINUS), down)`

Makes sense I guess; Minus on the Keypad is technically a different key, even though it results in the same character on solo press (Shift+Keypad Minus doesn't give you an underscore, unlike the one adjacent to Backspace).

**Comment by gregod** [comment]
I finally had time to fix this and just released v1.1.0. This should resolve both the capital letter and numpad issue! 

Could you please confirm that everything is working as expected with your setup ?

**Comment by GrandpaVape** [comment]
IT WORKS!

I can confirm this addon is now working with my wired Realinn USB 2D Barcode Scanner, accepting capital letters as well as dashes! I imagine it will work with other scanners too (as well as other codes besides the 1D linear codes I've been using), and I will test more at some point (whenever I have the chance to grab some more from work ;P). If and when I do, I can report my findings!

Thank you so much for your debugging help, @gregod! <3
